### PR TITLE
Batch norm

### PR DESCRIPTION
**Pull request by YuntianLan**
Added Batch Norm layer, as well as a test at examples/MNIST\ BatchNorm.jl for benchmark.

### CODE REVIEW

**Comment by stevenygd** [comment]
Need unit testing file and benchmark file.

File: src/layers/BatchNormLayer.jl [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    broadcast!(*,sq,xmu,xmu)
+    var = mean(sq,1)
+    l.var = deterministics ? l.var : var
+    # Normalize data
+    sqrtvar = Array{Float64}(1,D)
+    broadcast!(^,sqrtvar,var,0.5)
+    broadcast!(+,sqrtvar,sqrtvar,l.eps)
+    l.sqrtvar = deterministics ? l.sqrtvar : sqrtvar
+    ivar = Array{Float64}(1,D)
+    broadcast!(/,ivar,1,sqrtvar)
```
Comment (stevenygd) [inline src/layers/BatchNormLayer.jl:98]: it seems like you haven't used the epsilon doe numerical stabiity, have you checked that whether this division will create nan?

Comment (stevenygd) [reply src/layers/BatchNormLayer.jl:98]: Never mind , I saw it, but it seems like you are the epsilon after the square root, not before, so you might want to adjust the value of epsilon for that
